services/mizu/pipeline.go

```diff
@@ -6,20 +6,41 @@ import (
 	"github.com/sdslabs/gasper/types"
 )
 
-func buildPipeline(dockerImage string, confGenerator func(string, string) string) func(*types.ApplicationConfig) types.ResponseError {
-	return func(app *types.ApplicationConfig) types.ResponseError {
-		app.SetDockerImage(dockerImage)
-		app.SetConfGenerator(confGenerator)
-		return api.SetupApplication(app)
-	}
+// applicationHandler is a struct for managing the creation of a new application of a specific language
+type applicationHandler struct {
+	image         string
+	confGenerator func(string, string) string
 }
 
-var pipeline = map[string]func(*types.ApplicationConfig) types.ResponseError{
-	"nodejs":  buildPipeline(configs.ImageConfig.Nodejs, nil),
-	"php":     buildPipeline(configs.ImageConfig.Php, configs.CreatePHPContainerConfig),
-	"python2": buildPipeline(configs.ImageConfig.Python2, nil),
-	"python3": buildPipeline(configs.ImageConfig.Python3, nil),
-	"golang":  buildPipeline(configs.ImageConfig.Golang, nil),
-	"ruby":    buildPipeline(configs.ImageConfig.Ruby, nil),
-	"static":  buildPipeline(configs.ImageConfig.Static, configs.CreateStaticContainerConfig),
+// create handles the creation of a new application
+func (handler *applicationHandler) create(app *types.ApplicationConfig) types.ResponseError {
+	app.SetDockerImage(handler.image)
+	app.SetConfGenerator(handler.confGenerator)
+	return api.SetupApplication(app)
+}
+
+var pipeline = map[string]*applicationHandler{
+	"nodejs": {
+		image: configs.ImageConfig.Nodejs,
+	},
+	"python2": {
+		image: configs.ImageConfig.Python2,
+	},
+	"python3": {
+		image: configs.ImageConfig.Python3,
+	},
+	"golang": {
+		image: configs.ImageConfig.Golang,
+	},
+	"ruby": {
+		image: configs.ImageConfig.Ruby,
+	},
+	"php": {
+		image:         configs.ImageConfig.Php,
+		confGenerator: configs.CreatePHPContainerConfig,
```

Comment:
I hadn't notices it before but why aren't we using php dev server for this?

Reply:
nginx + php-fpm can handle more throughput than the PHP dev server which is single-threaded 
High throughput = better UX